### DISCUSSION TITLE
Add option for ranks.yml to read /cplaytime data

### DISCUSSION THREAD

**Discussion by thenameisKO** [Zrips/CMI#3915] [Feature Requests]
In the context of servers that have ranking systems that read playtime data but also do map resets once every few years, my staff and I believe having the option to have ranks.yml read data from /cplaytime would be beneficial.  From what we have seen during our most recent map reset, ranks.yml's "playtime" option was reading players' time online from the playerdata files in the world directory.  We cannot continue using those old files as they contain inventory data that is no longer needed and I was unable to find a way to get the CMI database to read in the data (even with PlayTimeFromStats set to "false" in config.yml).  We did notice that /cplaytime still maintained accurate data, but ranks.yml has no option to read that data right now.

In the interest of continuing to read from CMI's database and not have to rely on world data files for player info (other than inventories), I would like to request the option of ranks.yml having an option to read /cplaytime data for the ranking system.

**Comment by CloudeLecaw** [comment]
These "old files" are still used fir most things even after 10 years, statusitcs, values, Charakter, inventory, position, enderchest. All that has to do with Minecraft related to a player is saved there. Sometimes it helps for server owner to use cmi's purge system to delet outdated player files to get away from the million player that played 100years ago on the servers

> cplaytime (playerName)
> Detailed playtime in GUI format
> 
> Keep in mind that this is separate from playtime command and can show slightly different playtimes as it uses different engine to tract it

Looks like you can more trust into player time from playerfile ;) 
Cmi also give you the option to save playerfile somewhere else, but this value is also just read frim the playerfile ;-)

**Comment by thenameisKO** [comment]
With all due respect, that does not provide much help as we do not keep the same map over several years and do not want players to maintain inventories, positions, enderchest data, etc. when we reset a map.  The only concern I have is the playtime data, which is why I have created my request.

**Comment by Zrips** [comment]
@thenameisKO 
cplaytimne is some what experimental feature and it was created to have some what more detailed tracking of players playtime where you could check players playtime by an hour basis. Its not perfect and its working completely separately from general players playtime tracking server itself does. 
At this point what i can think of as a best possible solution would be a statistics wipe command where we could choose which stats we want to keep and wipe everything else from all player files. Then you could simply set for players playtime to be taken from stats instead of players data file and you would have continuing player playtime calculation as it should be

**Comment by thenameisKO** [comment]
Apologies for the late response on this; life got in the way a bit.  That would be a good option for the future, but I would greatly appreciate some sort of solution that both helps with my current issue and can help in the future as well.  Based on the schema of CMI's database, is there any field in it that the ranks.yml file can read?  I can see that there's a "playtime" table as well as a field for "TotalPlayTime" in the "players" table.
